Lib/fnmatch.py

```diff
@@ -54,17 +54,24 @@ def _compile_pattern(pat):
 def filter(names, pat):
     """Construct a list from those elements of the iterable NAMES that match PAT."""
     result = []
-    pat = os.path.normcase(pat)
-    match = _compile_pattern(pat)
+    pats = pat if isinstance(pat, tuple) else (pat,)
```

Comment:
My advice is to allow any iterable, not just tuple:

    pats = (pat,) if isinstance(pat, str) else pat  # Assume ``pat`` is an iterable